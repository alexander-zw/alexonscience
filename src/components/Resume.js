/**
 * This component contains my resume.
 *
 * This page is for anyone who came to learn about me personally, including
 * who I am, my skills and interests, and some of my projects that can't be
 * accessed directly on my website.
 */
import React from "react";
import { Link } from "react-router-dom";
import "../styles/index.css";
import "../styles/Resume.css";
import alexander_wu from "../images/alexander_wu.jpg";
import MetaTags from "./subcomponents/MetaTags";

function Resume() {
    const thesisLink = "https://www2.eecs.berkeley.edu/Pubs/TechRpts/2021/EECS-2021-102.html";

    // "contents" is not including heading.
    const sections = [
        {
            id: "education",
            title: "Education",
            contents: (
                <div>
                    <p>
                        Thesis:{" "}
                        <a href={thesisLink}>
                            Optimizations and Improvements to Cryptographic Libraries for zkSNARKs
                        </a>
                    </p>
                </div>
            ),
        },
        {
            id: "projects",
            title: "Projects",
            contents: (
                <div>
                    <p>
                        See my <a href="https://github.com/alexander-zw">GitHub</a> for more.
                    </p>
                    <p>
                        <strong>Guavabots</strong>
                        <br />
                        NP-complete graph problem with greedy algorithms and integer linear
                        programming, Python
                    </p>
                    <p>
                        <strong>Pintos</strong>
                        <br />
                        Operating system file storage with FFS, priority thread scheduling, and
                        process execution, C
                    </p>
                    <p>
                        <strong>Secure File Storage</strong>
                        <br />
                        Cryptographically encrypted and signed file storage and sharing system, Go
                    </p>
                    <p>
                        <strong>The Labyrinth</strong>
                        <br />
                        2D maze game with world generation and game mechanics, Java
                    </p>
                    <p>
                        <strong>BearMaps</strong>
                        <br />
                        Interactive navigation application similar to Google Maps, uses graphs and
                        A* search, Java
                    </p>
                    <p>
                        <strong>Stereo Vision</strong>
                        <br />
                        Detecting depth with two different images, quad tree compression, C and
                        RISC-V
                    </p>
                </div>
            ),
        },
    ];

    const tableOfContents = sections.map((sect, index) => (
        <li className="contents-li" key={index}>
            <a href={`#${sect.id}`} className="contents-item">
                {sect.title}
            </a>
        </li>
    ));

    const content = sections.map((sect, index) => (
        <section id={sect.id} className="section" key={index}>
            <h2>{sect.title}</h2>
            {sect.contents}
        </section>
    ));

    return (
        <div className="outer-container top-margin bottom-margin">
            <MetaTags path="/resume" />

            <nav className="contents-nav text-div">
                <ol className="contents-ol">{tableOfContents}</ol>
            </nav>

            <div className="text-div">
                <img className="photo" src={alexander_wu} alt={"Alexander Wu"} />

                <span className="name-title">Alexander Wu</span>

                <p>
                    Hello! I am not currently seeking a new position, but you can{" "}
                    <Link to="/contact">contact me</Link> for other matters. I{"'"}m a software
                    developer who expresses his love for science through my YouTube channel, among
                    other things. I{"'"}m passionate about education and enjoy doing volunteering
                    work. In my spare time, I read novels, work on projects, practice Yoga, and ride
                    my bike everywhere.
                </p>

                {content}
            </div>
        </div>
    );
}

export default Resume;
