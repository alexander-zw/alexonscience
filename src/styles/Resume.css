.outer-container {
    display: grid;
    grid-template-columns: 20% auto;
    grid-template-areas: "nav content";
}

.contents-nav {
    position: -webkit-sticky; /* Safari earlier versions. */
    position: sticky;
    top: 3.2em;
    align-self: start;
}

.contents-ol {
    list-style-type: none;
}

.contents-li {
    margin-top: 1em;
}

.contents-item {
    text-decoration: none;
    color: #636363;
}

.contents-item:hover {
    color: #008388;
}

.photo {
    width: 4em;
    height: 4em;
    border-radius: 2em;
    transform: translate(0, 0.9em);
    margin-right: 1.5em;
}

.name-title {
    padding-top: 0.2em;
    font-size: 3em;
    font-family: "Oleo Script", "Georgia", serif;
    background-image: linear-gradient(to right, #249141, #249c8a, #079cb3, #3464eb);
    -webkit-background-clip: text;
    -moz-background-clip: text;
    background-clip: text;
    color: transparent;
}

.section {
    /* This allows linking via table of contents to arrive at the right place. */
    padding-top: 1.5em;
}
