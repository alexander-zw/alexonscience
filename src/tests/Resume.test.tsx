import "@testing-library/jest-dom";

import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import Resume from "../components/Resume";

/**
 * Test for the Resume.tsx component.
 */
describe("Resume", () => {
    test("renders correctly", () => {
        // Resume has a Link, so we surround it with a router.
        const tree = renderer
            .create(
                <BrowserRouter>
                    <Resume />
                </BrowserRouter>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
