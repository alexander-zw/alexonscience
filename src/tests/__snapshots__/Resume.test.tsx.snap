// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume renders correctly 1`] = `
<div
  className="outer-container top-margin bottom-margin"
>
  <nav
    className="contents-nav text-div"
  >
    <ol
      className="contents-ol"
    >
      <li
        className="contents-li"
      >
        <a
          className="contents-item"
          href="#education"
        >
          Education
        </a>
      </li>
      <li
        className="contents-li"
      >
        <a
          className="contents-item"
          href="#projects"
        >
          Projects
        </a>
      </li>
    </ol>
  </nav>
  <div
    className="text-div"
  >
    <img
      alt="Alexander Wu"
      className="photo"
      src="alexander_wu.jpg"
    />
    <span
      className="name-title"
    >
      Alexander Wu
    </span>
    <p>
      Hello! I am not currently seeking a new position, but you can
       
      <a
        href="/contact"
        onClick={[Function]}
      >
        contact me
      </a>
       for other matters. I
      '
      m a software developer who expresses his love for science through my YouTube channel, among other things. I
      '
      m passionate about education and enjoy doing volunteering work. In my spare time, I read novels, work on projects, practice Yoga, and ride my bike everywhere.
    </p>
    <section
      className="section"
      id="education"
    >
      <h2>
        Education
      </h2>
      <div>
        <p>
          Thesis:
           
          <a
            href="https://www2.eecs.berkeley.edu/Pubs/TechRpts/2021/EECS-2021-102.html"
          >
            Optimizations and Improvements to Cryptographic Libraries for zkSNARKs
          </a>
        </p>
      </div>
    </section>
    <section
      className="section"
      id="projects"
    >
      <h2>
        Projects
      </h2>
      <div>
        <p>
          See my 
          <a
            href="https://github.com/alexander-zw"
          >
            GitHub
          </a>
           for more.
        </p>
        <p>
          <strong>
            Guavabots
          </strong>
          <br />
          NP-complete graph problem with greedy algorithms and integer linear programming, Python
        </p>
        <p>
          <strong>
            Pintos
          </strong>
          <br />
          Operating system file storage with FFS, priority thread scheduling, and process execution, C
        </p>
        <p>
          <strong>
            Secure File Storage
          </strong>
          <br />
          Cryptographically encrypted and signed file storage and sharing system, Go
        </p>
        <p>
          <strong>
            The Labyrinth
          </strong>
          <br />
          2D maze game with world generation and game mechanics, Java
        </p>
        <p>
          <strong>
            BearMaps
          </strong>
          <br />
          Interactive navigation application similar to Google Maps, uses graphs and A* search, Java
        </p>
        <p>
          <strong>
            Stereo Vision
          </strong>
          <br />
          Detecting depth with two different images, quad tree compression, C and RISC-V
        </p>
      </div>
    </section>
  </div>
</div>
`;
